### ISSUE TITLE
Caddy should use self-signed certificates for .home.arpa TLDs 

### ISSUE THREAD

**Issue by blissd** [caddyserver/caddy#5800]
### Actual Behaviour
The [.home.arpa](https://www.rfc-editor.org/rfc/rfc8375.html) TLD is reserved for internal use on home networks. However, when I configure Caddy with a .home.arpa domain it tries to auto-generate a certificate from letsencrypt and zerossl, which must and does fail.

### Expected Behavour
Caddy should not try to use letsencrypt or zerossl to generate a SSL certificate for .home.arpa domains. Instead, Caddy should generate a self-signed certificate, like it does for `localhost` domains and IP addresses.

**Comment by mholt** [comment]
That's probably true. Didn't think anyone actually used that. :upside_down_face:

**Comment by blissd** [comment]
I've been looking for the "correct" TLD for my home network for a long time... and now I've found it I want to use it 😆

**Comment by mholt** [comment]
I did this in https://github.com/caddyserver/certmagic/commit/66685874a86f20f433d38d00e2ab8a9250522b0d, and it'll go out the next time we upgrade CertMagic I suppose.

**Comment by dev-nicolaos** [comment]
Is there a reason a new version of certmagic hasn't been released since this work was done? Seems like it was getting regular releases over the summer, but there hasn't been one in the last couple months despite a good number of commits.

**Comment by francislavoie** [comment]
We've never released with any regularity, only when a release is needed/warranted.

But more specifically, Matt just had a premature baby which added complications to this project since he has less time for work.

We'll release 2.7.6 soon (hopefully in the next week or two).

**Comment by mholt** [comment]
Yep. Sorry. It's high on my priority list -- first is baby who just came home this week, then getting 2.7.6 out (which implies a new CertMagic tag) and then new website.

Feel free to use the latest commit in the meantime. Hopefully soon!!

**Comment by WoutervanZijl** [comment]
Hello there I am using caddy and don't understand how to use my home.arpa domain in the caddyfile or isnt ready yet? maybe i do something wrong. Fresh install and inside the caddy file there is the fist line with port number do i changed there anything? or remove everything and just start with reverse proxy? keep getting errors when reloading. Cheers for such a wonderful peace of software! Makes life alot easier!

**Comment by mholt** [comment]
I think if you use the upcoming release it should work.

**Comment by WoutervanZijl** [comment]
> I think if you use the upcoming release it should work. Try one of the latest artifacts: https://github.com/caddyserver/caddy/actions/runs/8677592492

I have to say I am completely new to what youre a suggestion! I do want to try but dont know where to begin:)

So when the next release is there it will be be implemented?

**Comment by mholt** [comment]
Just download the file for your platform and run it -- it's like you would do for a release binary, it's just from CI instead.

**Comment by ffuhrnew** [comment]
You cannot use *.home.arpa - you shall use *.yourname.home.arpa.
You could create a Caddyfile with:
caddy.yourname.home.arpa  {
        root * /usr/share/caddy
        file_server
}
Add the root ca from /var/lib/caddy/.local/share/caddy/pki/authorities/local/root.crt as trusted ca in your browser.
